makefile

```diff
@@ -38,6 +38,22 @@ CXX = $(CC)
 ##
 -include make/detect_os
 
+## FIXME(dl): move to appropriate place later
+## GPU settings
+
+ifdef STAN_GPU
+  CXXFLAGS += -isystem $(OPENCL) -DSTAN_GPU -DSTAN_DEVICE_CPU
+  ifeq ($(shell echo $(TRAVIS_OS_NAME)),linux)
+    LDLIBS+=-L/home/travis/AMDAPPSDK/lib/x86_64/
+  endif
+
+  ifeq ($(OS_TYPE),mac)
+    LDFLAGS += -framework OpenCL
+  else
+    LDFLAGS += -lOpenCL
+  endif
+endif
```

Comment:
We need to set a STAN_GPU env variable in Jenkins. Does someone with actual access to the machine do that or is this something that is changed in the Jenkinsfile? I have to read more about Jenkins before I feel comfortable touching that file

Reply:
You can set it in the Jenkinsfile for specific tests. Do you need an environment variable or a compiler flag? A compiler flag is set like this: https://github.com/stan-dev/math/blob/develop/Jenkinsfile#L156

Reply:
Since we are using `ifdef STAN_GPU` I believe we need an environment variable. I'll write a function like `setupCC()` in the Jenkinsfile that runs for each of the stages that needs the OpenCL code in `stage('Tests')`

Reply:
Also I think this should be moved to each `os_foo` files since we will want different options for each compilers / os. e.g. mac needs `-framework OpenCL` while Linux wants `-lOpenCL`. I believe clang++ wants either of those in `LDFLAGS` while `g++` wants those in `LDLIBS`

Reply:
If you're talking about using `ifdef` in the C++ preprocessor, that's a compiler flag it's looking for, I believe. So you should be able to do it the same way. 

And yeah that move sounds good

Reply:
I think the changes I made we're overridden and should be reverted back. We can rely on make variables instead of environment variables. That was done intentionally since we haven't used environment variables so far for our build process. 

Re: moving to os_*. Let's get it working properly first and then I can move it appropriately.

Reply:
I think the changes I made we're overridden and should be reverted back. We can rely on make variables instead of environment variables. That was done intentionally since we haven't used environment variables so far for our build process. 

Re: moving to os_*. Let's get it working properly first and then I can move it appropriately.

Reply:
> I think the changes I made we're overridden and should be reverted back.

Your changes from  112276a9 and eab81d160? Those are still here but

1. You had double double quotes `""` at the end of `MATRIX_EVAL` I assumed this was a whoopsie and made those a single double quote.  

2.  Using `LDLIBS` would not pass through when using `clang++` so I set these to `LDFLAGS`. Though I believe we need `LDLIBS` for `g++`

Reply:
> Re: moving to os_*. Let's get it working properly first and then I can move it appropriately.

I concur

Reply:
Yeah, you're absolutely right. I have no idea how I thought the changes I made we're backed out.

I'll take a closer look later tonight.

Reply:
>> Re: moving to os_*. Let's get it working properly first and then I can move it appropriately.

> I concur

After messing around with this I ended up not concuring. It was easier to work with by breaking it up into the os_type files. If this messes up something else you have going on at the moment I'll put it all back in the makefile